containerstats-pipeline.md

```diff
@@ -0,0 +1,47 @@
+# ContainerStats Pipeline
+
+The ContainerStats pipeline builds statistical representations of individual
+containers' resource usage over time.  The pipeline is part of the `cost-model` container.
+
+## APIs
+
+### Request right-sizing recommendation v2
+
+The primary user of ContainerStats pipeline data is v2 of the request right-
+sizing recommendation API. ContainerStats data is used for quantile-based
+recommendations. Review the doc for this feature [here](https://github.com/kubecost/docs/blob/main/api-request-right-sizing-v2.md).
+
+### Debugging
+
+There is an _unstable_, _unofficial_ API for introspecting pipeline data
```

Comment:
I don't know how I feel about this emphasis on the word unstable. Would this API still be considered in beta, or in testing? I think there are better word choices.

Reply:
I think it depends... if it's going to be something we push out as "beta", not fit for production usage and will sit unfinished for a long time, I think the emphasis is important, also, if there's any risk of us just dropping it randomly. If it's something we are committed to finishing and making stable, officially supporting within a sane timeframe, I agree that softer language is okay.

Can using the unfinished API cause any sort of downtime or data loss?

Reply:
I can get behind informing the user that a feature is still being tested/developed, but calling it unstable may make them wary of even looking into it, so I think we should elaborate what the specific harm/faults could be here.

Reply:
My backstory is that at my last job, we launched MySQL servers under "beta" and they sat with no further dev for 2-3 years, customers sometimes permanently lost data and many used it in production. Bad business decisions on behalf of said employer, for sure, but I did fight hard to include additional unstable/unsafe language.

Not that our product is as mission critical and can open the customer to critical data loss ;) Just wanted to share my thought process, there's a balance there somewhere and if this API could be suddenly dropped or deemed unfeasible in the future, totally cool with erring on the side of non production language, imo

Reply:
My intent was to communicate that the API endpoint being discussed in this subsection (`/model/containerstats/quantiles`) does not have a **stable schema** and is **not supported as an official product feature**. Rather, it is a debugging-only endpoint mostly for use by support/solutions eng with users OR by eng during development.

Ideally, this would belong in an "internal-only" section of this document, but I'm not sure if we are able to do that right now.

For either/both of you, is there a rephrase you would suggest based on this additional context?

Reply:
I think the way you put it, not being supported currently as an official product feature" is probably better than "unstable." Is there a reason this section needs to be included at all however, as it's mainly for internal use?

Reply:
I'm not aware of an internal-only documentation location for this information that people actually check. Is there one?

Reply:
Also, updated phrasing.

Reply:
I don't know about an internal doc specifically, I'm just saying if we are having difficult wording the description of a non-official feature, we should consider not documenting the feature at all. Otherwise, if we think it should still be added to the public doc, I word prefer your description in this thread as a clarifier as opposed to saying "unstable." However, this is close to a non-issue so I really don't think it needs more discussion.

Reply:
I updated the description to use what I said earlier in the thread instead of unstable.